digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/CalibrationPointDAO.scala

```diff
@@ -79,23 +95,55 @@ object CalibrationPointDAO {
      """.as[CalibrationPoint].firstOption
   }
 
-  def fetchByRoadwayPoint(roadwayPointId:Long) : Option[CalibrationPoint] = {
+  def fetch(linkId: Long, startOrEnd: CalibrationPointLocation, roadwayPointId: Long): Option[CalibrationPoint] = {
     sql"""
-         SELECT CP.ID, ROADWAY_POINT_ID, LINK_ID, ROADWAY_NUMBER, RP.ADDR_M, START_END, TYPE, VALID_FROM, VALID_TO, CP.CREATED_BY, CP.CREATED_TIME
+         SELECT CP.ID, CP.ROADWAY_POINT_ID, CP.LINK_ID, ROADWAY_NUMBER, ADDR_M, CP.START_END, CP.TYPE, CP.VALID_FROM, CP.VALID_TO, CP.CREATED_BY, CP.CREATED_TIME
+         FROM CALIBRATION_POINT CP
+         JOIN ROADWAY_POINT RWP ON (RWP.id = CP.ROADWAY_POINT_ID)
+         WHERE CP.ROADWAY_POINT_ID = $roadwayPointId
+         AND CP.LINK_ID = $linkId
+         AND CP.START_END = ${startOrEnd.value}
+         AND CP.VALID_TO IS NULL
+     """.as[CalibrationPoint].firstOption
+  }
+
+  def fetchByRoadwayPointId(roadwayPointId: Long): Option[CalibrationPoint] = {
+    sql"""
+         SELECT CP.ID, ROADWAY_POINT_ID, LINK_ID, ROADWAY_NUMBER, RP.ADDR_M, START_END, CP.TYPE, VALID_FROM, VALID_TO, CP.CREATED_BY, CP.CREATED_TIME
          FROM CALIBRATION_POINT CP
          JOIN ROADWAY_POINT RP
           ON RP.ID = CP.ROADWAY_POINT_ID
          WHERE cp.roadway_point_id = $roadwayPointId
      """.as[CalibrationPoint].firstOption
   }
 
+  def fetchIdByRoadwayNumberSection(roadwayNumber: Long, startAddr: Long, endAddr: Long): Set[Long] = {
+    sql"""
+         SELECT CP.ID
+         FROM CALIBRATION_POINT CP
+         JOIN ROADWAY_POINT RP ON RP.ID = CP.ROADWAY_POINT_ID
+         WHERE RP.roadway_number = $roadwayNumber AND RP.ADDR_M BETWEEN $startAddr AND $endAddr AND CP.VALID_TO IS NULL
+     """.as[Long].list.toSet
+  }
+
+  def fetchIdByRoadwayPointIdWithJunctionDefined(roadwayPointId: Long, addr: Long): Set[Long] = {
+    sql"""
+         SELECT CP.ID
+         FROM CALIBRATION_POINT CP
+         JOIN ROADWAY_POINT RP ON RP.ID = CP.ROADWAY_POINT_ID
+         JOIN JUNCTION_POINT JP ON JP.ROADWAY_POINT_ID = RP.ID
+         JOIN JUNCTION J ON J.ID = JP.JUNCTION_ID
```

Comment:
Should here be also: `AND J.VALID_TO IS NULL AND J.END_DATE IS NULL`?

Reply:
No because we dont have the guarantee that the Junction didnt expired. 
And this query is to fetch roadway points related with expired junction points.

Will rename it.